### PR TITLE
WIP: Move OpenCommPort to api set

### PR DESCRIPTION
**Pull request by pjanotti**
It will fail until changes in buildtools (https://github.com/dotnet/buildtools/pull/1766) reach corefx.

### CODE REVIEW

File: src/Common/src/Interop/Windows/Interop.Libraries.cs [hunk 1]
```diff
@@ -30,5 +30,6 @@ internal static class Libraries
         internal const string WinHttp = "winhttp.dll";
         internal const string Ws2_32 = "ws2_32.dll";
         internal const string Zlib = "clrcompression.dll";
+
```
Comment (danmosemsft) [inline src/Common/src/Interop/Windows/Interop.Libraries.cs]: nit blank line

File: src/Common/src/Interop/Windows/mincore/Interop.OpenCommPort.cs [hunk 2]
```diff
@@ -9,9 +9,9 @@
 
 internal partial class Interop
 {
-    internal partial class KernelBase
+    internal partial class mincore
```
Comment (safern) [inline src/Common/src/Interop/Windows/mincore/Interop.OpenCommPort.cs]: should this be capitalized?

Comment (pjanotti) [reply src/Common/src/Interop/Windows/mincore/Interop.OpenCommPort.cs]: I am actually following what is already in the sources (about 4 other source files). If there is consensus in this regard I can take the opportunity to rename them all (scope of the change increases to System.Memory).

Comment (safern) [reply src/Common/src/Interop/Windows/mincore/Interop.OpenCommPort.cs]: I see. I was just wondering, I thought our guideline was to have classes capitilized but maybe this are exceptions as they are interop classes. cc: @stephentoub

File: src/Common/src/Interop/Windows/mincore/Interop.OpenCommPort.cs [hunk 3]
```diff
@@ -9,9 +9,9 @@
...
     {
-        [DllImport(Libraries.KernelBase, SetLastError = true)]
+        [DllImport(Libraries.CoreComm_L1_1_1, SetLastError = true)]
```
Comment (jkotas) [inline src/Common/src/Interop/Windows/mincore/Interop.OpenCommPort.cs]: This will not work on older Windows versions. Why are we doing this change?

Comment (pjanotti) [reply src/Common/src/Interop/Windows/mincore/Interop.OpenCommPort.cs]: This change was requested by Windows team. The API is only used by UWP and was introduced in RS3, for Windows non-Uap it uses kernel32!CreateFile.

Comment (jkotas) [reply src/Common/src/Interop/Windows/mincore/Interop.OpenCommPort.cs]: The Interop.OpenCommPort.cs  file should be compiled in for Uap only then.

Comment (pjanotti) [reply src/Common/src/Interop/Windows/mincore/Interop.OpenCommPort.cs]: It is via the wrapper in System.IO.Ports and the settings in the project:
https://github.com/dotnet/corefx/blob/c3e50ed1c017b36c5b8fe19a5dc9c499369eeb43/src/System.IO.Ports/src/System/IO/Ports/SerialStream.Uap.cs#L13
https://github.com/dotnet/corefx/blob/a989d4f91cf4208b7a2481d120498a581328eb1d/src/System.IO.Ports/src/System.IO.Ports.csproj#L139

Comment (jkotas) [reply src/Common/src/Interop/Windows/mincore/Interop.OpenCommPort.cs]: Could you please move `$(CommonPath)\Interop\Windows\mincore\Interop.OpenCommPort.cs` under this condition as well?

Comment (pjanotti) [reply src/Common/src/Interop/Windows/mincore/Interop.OpenCommPort.cs]: Yes, that makes things clear and explicit, added.

**Comment by jkotas** [comment]
Merged into https://github.com/dotnet/corefx/pull/24944 to fix build breaks

**Pull request closed without merging**